Automattic/wp-calypso
# composite-checkout: Support GSuite products

#### Changes proposed in this Pull Request

This updates composite checkout to support G Suite products. 

Requires D43415-code and D43492-code (already merged).

Review step:

![Screen Shot 2020-05-12 at 5 29 36 PM](https://user-images.githubusercontent.com/2036909/81747514-39809000-9476-11ea-9b58-694e27b6d156.png)

Inactive review step:

![Screen Shot 2020-05-12 at 5 26 47 PM](https://user-images.githubusercontent.com/2036909/81747521-3f767100-9476-11ea-90bf-95c2624bb4eb.png)

Contact fields:

![Screen Shot 2020-05-18 at 2 14 40 PM](https://user-images.githubusercontent.com/2036909/82246013-f741cd80-9911-11ea-8414-f443cdff8f54.png)

#### Testing instructions

To test the purchase flow:

- Add a domain to your cart and head to checkout (make sure you see composite checkout; you may need to add `?flags=composite-checkout-force` to the URL).
- You will see the G Suite upsell form. Add two email addresses.
- Verify that you see the G Suite product and both email addresses listed in the review.
- Complete the purchase with a test credit card and verify that the purchase was successful and the G Suite products were activated successfully.
- Repeat the purchase with PayPal since this goes through a different transaction endpoint.

To test the G Suite form fields:

- With a domain already purchased, add G Suite to your cart.
- Add at least one G Suite email address.
- Visit checkout (make sure you see composite checkout; you will need to add `?flags=composite-checkout-force` to the URL).
- Verify that step 2 of the checkout form shows fields for first name, last name, alternate email, country, and postal code.
- Verify that the fields are pre-filled with your domain contact information, particularly the alternate email field.
- Empty some fields or fill them with invalid characters and press "Continue".
- Verify that the step does not complete and that you see the invalid fields highlighted.
- Set the alternate email field to use the domain name you are adding G Suite to (eg: if the domain is example.com, set the alternate email field to `foobar@example.com`).
- Press "Continue" and verify that you see an error on the email field saying that you cannot use that domain.
- Set all fields to valid data and press "Continue".
- Verify that you are able to reach the final step.
- Complete the purchase and verify that it was successful.
